src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -788,6 +788,24 @@ private bool IsValidExtensionMethodSignature()
                 this.ParameterRefKinds.IsDefault && // No 'ref' or 'out'
                 !this.IsParams();
 
+        private bool IsValidUserDefinedOperatorIs()
+        {
+            foreach (var parameter in this.Parameters)
+            {
+                if (parameter.RefKind != ((parameter.Ordinal == 0) ? RefKind.None : RefKind.Out))
+                {
+                    return false;
+                }
+            }
+
+            return
+                (this.ReturnsVoid || this.ReturnType.SpecialType != SpecialType.System_Boolean) &&
+                !this.IsGenericMethod &&
+                !this.IsVararg &&
+                this.ParameterCount > 0 &&
```

Comment:
I'd rather not introduce a new and inconsistent style here. We currently have a complete separation between the PE symbol import import and semantic analysis, and a separate and different analysis for the categorization of operators. That is fine. For example, binary operators in PE do not require one of the arguments be the containing type, but that requirement is imposed in source. The only requirement for the relationship between the two is that the source is as strict or more strict than PE.